gulpfile.js

```diff
@@ -20,45 +21,88 @@ const styles = () => {
       autoprefixer()
     ]))
     .pipe(sourcemap.write("."))
-    .pipe(dest("source/css"))
+    .pipe(dest("dist/css"))
     .pipe(sync.stream());
 }
 
 exports.styles = styles;
 
-// Logo
+// HTML
 
-const logo = () => {
-  return src("source/logo/*.svg")
+const html = () => {
+  return src("source/*.html")
+    .pipe(htmlmin({
+      collapseWhitespace: true
+    }))
+    .pipe(dest("dist"));
+}
+
+// Images
+
+const images = () => {
+  return src("source/img/**/*.{png,jpg,svg}")
     .pipe(imagemin([
+      imagemin.mozjpeg({
+        progressive: true
+      }),
+      imagemin.optipng({
+        optimizationLevel: 3
+      }),
       imagemin.svgo()
     ]))
-    .pipe(dest("source/icons"))
+    .pipe(dest("source/img"))
+}
+
+exports.images = images;
+
+// Logo
+
+const logo = () => {
+  return src("source/img/logo/*.svg")
+    .pipe(dest("dist/icons"))
 }
 
 exports.logo = logo;
 
 // Svg stack
 
 const svgstack = () => {
-  return src("source/img/**/*.svg")
+  return src("source/icons/*.svg")
```

Comment:
При том, что лучше избегать подпапок, но и предусмотреть в таске случай их наличия всё-таки стоит.
```suggestion
  return src("source/icons/**/*.svg")
```